pkg/analyze/cluster_container_statuses.go

```diff
@@ -0,0 +1,252 @@
+	"github.com/replicatedhq/troubleshoot/pkg/constants"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type AnalyzeClusterContainerStatuses struct {
+	analyzer *troubleshootv1beta2.ClusterContainerStatuses
+}
+
+type podsWithContainers map[string]struct {
+	name              string
+	namespace         string
+	containerStatuses []corev1.ContainerStatus
+}
+
+type matchedContainerInfo struct {
+	Namespace     string
+	PodName       string
+	ContainerName string
+	Ready         bool
+	RestartCount  int32
+	Message       string
+}
+
+func (a *AnalyzeClusterContainerStatuses) Title() string {
+	if a.analyzer.CheckName != "" {
+		return a.analyzer.CheckName
+	}
+	return "Cluster Container Status"
+}
+
+func (a *AnalyzeClusterContainerStatuses) IsExcluded() (bool, error) {
+	return isExcluded(a.analyzer.Exclude)
+}
+
+func (a *AnalyzeClusterContainerStatuses) Analyze(getFile getCollectedFileContents, findFiles getChildCollectedFileContents) ([]*AnalyzeResult, error) {
+	// get all pod list files from clusterResources collector directory
+	excludeFiles := []string{}
+	podListFiles, err := findFiles(filepath.Join(constants.CLUSTER_RESOURCES_DIR, constants.CLUSTER_RESOURCES_PODS, "*.json"), excludeFiles)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read collected pods")
+	}
+
+	// get pods matched analyzer filters
+	pods, err := a.getPodsMatchingFilters(podListFiles)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get pods matching filters")
+	}
+
+	results, err := a.analyzeContainerStatuses(pods)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to analyze container statuses")
+	}
+
+	return results, nil
+}
+
+func (a *AnalyzeClusterContainerStatuses) getPodsMatchingFilters(podListFiles map[string][]byte) (podsWithContainers, error) {
+	matchedPods := podsWithContainers{}
+
+	for fileName, fileContent := range podListFiles {
+		// pod list fileName is the namespace name, e.g. default.json
+		currentNamespace := strings.TrimSuffix(filepath.Base(fileName), ".json")
+		selectedNamespaces := a.analyzer.Namespaces
+		if len(selectedNamespaces) > 0 {
+			if !slices.Contains(selectedNamespaces, currentNamespace) {
+				continue
+			}
+		}
+
+		// filter pods by namespace
+		var podsMatchedNamespace []corev1.Pod
+		var podList corev1.PodList
+		if err := json.Unmarshal(fileContent, &podList); err != nil {
+			var pods []corev1.Pod
+			// fallback to old format
+			if err := json.Unmarshal(fileContent, &pods); err != nil {
+				return nil, errors.Wrapf(err, "failed to unmarshal pods list for namespace %s", currentNamespace)
+			}
+			podsMatchedNamespace = append(podsMatchedNamespace, pods...)
+		} else {
+			podsMatchedNamespace = append(podsMatchedNamespace, podList.Items...)
+		}
+
+		// filter pods by container restart count
+		for _, pod := range podsMatchedNamespace {
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.RestartCount < int32(a.analyzer.RestartCount) {
```

Comment:
just wondering should we set a default RestartCount, like 5. Some customer may not set RestartCount, since they don't know and until saw it. I would suggest to have a default one if not specify it

Reply:
Currently, we have `RestartCount` default set to `0`, which ensures that all containers are monitored without missing any data. Setting a default value of 5 could inadvertently exclude containers that restart fewer times, potentially missing important information for users who rely on monitoring those instances. I would say customer can refer to document to set desired number of `RestartCount` based on their requirements. E.g. OOMKills detect